config/admin-functions/choose-vx-machine-id.sh

```diff
@@ -6,12 +6,12 @@ set -euo pipefail
 
 while true; do
   read -p "${VX_MACHINE_TYPE} Machine ID (e.g. 0012): " MACHINE_ID
-  if [[ "${MACHINE_ID}" =~ ^[0-9]+$ ]]; then
+  if [[ "${MACHINE_ID}" =~ ^[-0-9A-Z]+$ ]]; then
```

Comment:
does the first `-` need to be backslashed? I hate regexps.

Reply:
> does the first `-` need to be backslashed?

[No](https://regexr.com/63sb7):

<img width="896" alt="image" src="https://user-images.githubusercontent.com/1938/129637116-99faa989-2f92-4470-9a9a-195024f66c16.png">

> I hate regexps.

This is expected behavior.